server/models.py

```
@@ -0,0 +1,193 @@
+        return f'<Booking for {self.patient.name} at {self.clinic.name}>'
+
+
+
+class User(db.Model):
+    id = db.Column(db.Integer, primary_key=True)
+    username = db.Column(db.String(80), unique=True, nullable=False)
+    password_hash = db.Column(db.String(128), nullable=False)
+    role = db.Column(db.String(20), nullable=False, default='patient')  # 'admin', 'clinic', or 'patient'
+
+    def set_password(self, password):
+        self.password_hash = generate_password_hash(password).decode('utf-8')
```

Ruff (0.11.9)

Undefined name `generate_password_hash`

(F821)